cmd/bnsd/client/client.go

```diff
@@ -392,6 +394,33 @@ func walletKeyToAddr(key []byte) weave.Address {
 	return key[5:]
 }
 
+type CurrenciesResponse struct {
+	Height     int64
+	Currencies map[string]currency.TokenInfo
+}
+
+func (b *BnsClient) ListCurrencies() (*CurrenciesResponse, error) {
+	resp, err := b.AbciQuery("/tokens?prefix", nil)
+	if err != nil {
+		return nil, err
```

Comment:
I think it's better to wrap the error, so that it's easier to track where it comes from.